openfl/component/director/director.py

```diff
@@ -205,51 +432,12 @@ def get_envoys(self) -> list:
 
         return envoy_infos
 
-    async def _run_aggregator_in_workspace(
-            self,
-            *,
-            experiment_name: str,
-            data_file_name: Path,
-            **kwargs
-    ) -> None:
-        """Run aggregator in a workspace."""
-        with ExperimentWorkspace(experiment_name, data_file_name):
-            await self._run_aggregator(experiment_name=experiment_name, **kwargs)
+    async def run_background_tasks(self):
+        """Run director's background tasks."""
+        loop = asyncio.get_event_loop()
+        loop.create_task(self._monitor_experiment_task())
 
-    async def _run_aggregator(
-            self,
-            *,
-            experiment_sender,
-            initial_tensor_dict,
-            experiment_name,
-            collaborator_names,
-            plan_path='plan/plan.yaml'
-    ) -> None:
-        """Run aggregator."""
-        plan = Plan.parse(plan_config_path=Path(plan_path))
-        plan.authorized_cols = list(collaborator_names)
-
-        logger.info('🧿 Starting the Aggregator Service.')
-        aggregator_server = plan.interactive_api_get_server(
-            tensor_dict=initial_tensor_dict,
-            root_certificate=self.root_certificate,
-            certificate=self.certificate,
-            private_key=self.private_key,
-            tls=self.tls,
-        )
-        self.experiment_stash[experiment_sender][experiment_name] = aggregator_server
-
-        grpc_server = aggregator_server.get_server()
-        grpc_server.start()
-        logger.info('Starting Aggregator gRPC Server')
-
-        try:
-            while not aggregator_server.aggregator.all_quit_jobs_sent():
-                # Awaiting quit job sent to collaborators
-                await asyncio.sleep(10)
-        except KeyboardInterrupt:
-            pass
-        finally:
-            grpc_server.stop(0)
-            # Temporary solution to free RAM used by TensorDB
-            aggregator_server.aggregator.tensor_db.clean_up(0)
+    async def _monitor_experiment_task(self):
+        while True:
+            await self.experiments_registry.set_next_experiment()
```

Comment:
Maybe we replace these to methods with another one? `run_next_experiment` for example.

Reply:
`run_next_experiment` looks like it should stop current experiment and start next.

Reply:
`set_next_experiment` and `start_active` was combined to `run_next_experiment`